python/examples/advanced_agents/Browser_reviews_scraper/browser_tool_outlook.py

```diff
@@ -0,0 +1,50 @@
+from langchain.agents import create_openai_functions_agent, AgentExecutor
+from langchain import hub
+from langchain_openai import ChatOpenAI
+from composio_langchain import ComposioToolSet
+llm = ChatOpenAI()
+import asyncio
+import logging
+
+prompt = hub.pull("hwchase17/openai-functions-agent")
+
+composio_toolset = ComposioToolSet(api_key="fcau1ynif45lumo8txt5o", connected_account_ids={})
+tools = composio_toolset.get_tools(actions=['BROWSER_TOOL_GET_PAGE_DETAILS','BROWSER_TOOL_GOTO_PAGE', 'BROWSER_TOOL_SCROLL_PAGE', 'BROWSER_TOOL_REFRESH_PAGE', 'GMAIL_SEND_EMAIL'])
+
+agent = create_openai_functions_agent(llm, tools, prompt)
+agent_executor = AgentExecutor(
+    agent=agent, 
+    tools=tools, 
+    verbose=True,
+    handle_parsing_errors=True
+)
+def get_reviews(url):
```

Comment:
Input Validation: The URL input is not validated before processing. Add URL validation to prevent processing invalid URLs:

```python
from urllib.parse import urlparse

def validate_url(url: str) -> bool:
    """Validate if the given URL is properly formatted."""
    try:
        result = urlparse(url)
        return all([result.scheme, result.netloc])
    except Exception:
        return False

def get_reviews(url: str) -> None:
    if not validate_url(url):
        raise ValueError(f"Invalid URL format: {url}")
    # Continue with processing...